python/ray/tune/execution/trial_runner.py

```diff
@@ -147,51 +149,61 @@ def __init__(
 
         self._max_pending_trials = _get_max_pending_trials(self._search_alg)
 
+        self._storage = storage
         self._sync_config = sync_config or SyncConfig()
 
-        # Rename for better code readability
-        local_experiment_path, remote_experiment_path = _split_remote_local_path(
-            experiment_path, None
-        )
-
-        # Derive experiment dir name from local path
-        if not experiment_dir_name and local_experiment_path:
-            # Maybe derive experiment dir name from local storage dir
-            experiment_dir_name = Path(local_experiment_path).name
-        elif not experiment_dir_name:
-            experiment_dir_name = DEFAULT_EXPERIMENT_NAME
-
-        # Set default experiment dir name
-        if not local_experiment_path:
-            local_experiment_path = str(
-                Path(_get_defaults_results_dir()) / experiment_dir_name
+        if _use_storage_context():
+            assert storage
+            self._experiment_dir_name = storage.experiment_dir_name
+            self._local_experiment_path = storage.experiment_cache_path
+            self._remote_experiment_path = storage.experiment_fs_path
```

Comment:
Note: These concepts of "local" and "remote" should be rebranded as "cache_path" and "storage_fs_path".